src/PositionManager.sol

```diff
@@ -60,80 +63,62 @@ contract PositionManager is
     }
 
     /// @param unlockData is an encoding of actions, params, and currencies
-    /// @return returnData is the endocing of each actions return information
     function modifyLiquidities(bytes calldata unlockData, uint256 deadline)
         external
         payable
         isNotLocked
         checkDeadline(deadline)
-        returns (bytes[] memory)
     {
-        // TODO: Edit the encoding/decoding.
-        return abi.decode(poolManager.unlock(unlockData), (bytes[]));
+        _executeActions(unlockData);
     }
 
-    function _unlockCallback(bytes calldata payload) internal override returns (bytes memory) {
-        (Actions[] memory actions, bytes[] memory params) = abi.decode(payload, (Actions[], bytes[]));
-
-        bytes[] memory returnData = _dispatch(actions, params);
-
-        return abi.encode(returnData);
+    function _handleAction(uint256 action, bytes calldata params) internal override {
+        if (action == Actions.INCREASE_LIQUIDITY) {
+            _increase(params);
+        } else if (action == Actions.DECREASE_LIQUIDITY) {
+            _decrease(params);
+        } else if (action == Actions.MINT_POSITION) {
+            _mint(params);
+        } else if (action == Actions.CLOSE_CURRENCY) {
+            _close(params);
+        } else if (action == Actions.BURN_POSITION) {
+            // Will automatically decrease liquidity to 0 if the position is not already empty.
+            _burn(params);
```

Comment:
dont need to change it now, and its more a discussion bit --

I think decoding the parameters before making the internal calls makes the internal function signatures much more readable (and easier to reference)

Reply:
I think this makes gas go down quite a bit though bc we're just passing calldata